docs/tuning.md

```diff
@@ -232,8 +232,12 @@ temporary objects created during task execution. Some steps which may be useful
   value of the JVM's `NewRatio` parameter. Many JVMs default this to 2, meaning that the Old generation 
   occupies 2/3 of the heap. It should be large enough such that this fraction exceeds `spark.memory.fraction`.
   
-* Since 4.0.0, Spark uses JDK 17 by default, which also makes the G1GC garbage collector the default. Note that with
-  large executor heap sizes, it may be important to increase the G1 region size with `-XX:G1HeapRegionSize`.
+* Since 4.0.0, Spark uses JDK 17 by default, which also makes the G1GC garbage collector the default,
+  Some applications may encounter out-of-memory (OOM) issues or experience slower performance.
+  To mitigate these problems:
+  - Increase region size with `-XX:G1HeapRegionSize=<size>` when using large executor heap sizes.
+  - Optimize GC locker behavior with `-XX:+UnlockDiagnosticVMOptions -XX:GCLockerRetryAllocationCount=100`.
+  - Consider switching to ParallelGC by specifying `-XX:+UseParallelGC` for better throughput with large workloads.
```

Comment:
JDK21 >=21.0.4 + Parallel GC may exceed memory, causing Executor to be killed by YARN.

---

8328744: Parallel: Parallel GC throws OOM before heap is fully expanded

https://bugs.openjdk.org/browse/JDK-8328744

https://bugs.openjdk.org/browse/JDK-8338920

An existing bug may have prevented the full usage of the Java heap allotted with the command line flag -Xmx. That bug has been fixed with JDK-8328744. As a side effect of this fix, installations may experience an increased heap usage when using Parallel GC. Customers should, if necessary, adjust the maximum heap size.